src/app/ui-kit/components/radio-group/radio-group.component.ts

```diff
@@ -0,0 +1,38 @@
+import { Component, EventEmitter, forwardRef, Input, OnInit, Output } from '@angular/core';
+import { ControlValueAccessor, NG_VALUE_ACCESSOR } from '@angular/forms';
+
+@Component({
+  selector: 'app-ui-radio-group',
+  templateUrl: './radio-group.component.html',
+  providers: [
+    {
+      provide: NG_VALUE_ACCESSOR,
+      useExisting: forwardRef(() => RadioGroupComponent),
+      multi: true,
+    },
+  ],
+})
+export class RadioGroupComponent implements ControlValueAccessor {
+  @Input() options: string[] = [];
+  value: string | null = null;
+  onChange: (value: string) => void = () => {};
+  onTouched: () => void = () => {};
+  
+  selectOption(option: string) {
+    this.writeValue(option);
+    this.onChange(option);
+    this.onTouched();
+  }
+
+  writeValue(value: string): void {
+    this.value = value;
+  }
+
+  registerOnChange(fn: (value: string) => void): void {
+    this.onChange = fn;
+  }
+
+  registerOnTouched(fn: () => void): void {
+    this.onTouched = fn;
+  }
+}
```

Comment:
this approach with implementing ControlValueAccessor looks to complicated. I might not see the whole picture right now, let it be, but let's revise this approach when you will be working with a real data. 

You can pass control to your input with type="radio" like
`[formControl]="yourCustomControl"`
and then listen to control value changes, or set value using yourCustomControl.setValue, and just get form data in the parent component.